### PR TITLE
LGA-3352 - Add safeguarding question

### PR DESCRIPTION
**Pull request by said-moj**
## What does this pull request do?

Add safeguarding question.

## Any other changes that would benefit highlighting?

Intentionally left blank.

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: app/categories/forms.py [hunk 1]
```diff
@@ -23,11 +24,48 @@ class QuestionForm(Form):
...
+    submit = SubmitField(_("Continue"), widget=GovSubmitInput())
+
+    def __init__(self, *args, **kwargs):
+        if "category" in kwargs:
+            self.category = kwargs.pop("category")
+        super().__init__(*args, **kwargs)
+
+
+class SafeguardingQuestionForm(QuestionForm):
+    """Base form for question pages with configurable routing based on answers."""
```
Comment (BenMillar-MOJ) [inline app/categories/forms.py:43]: Can these duplicate comments be removed please, just to keep this class clean.

File: app/categories/widgets.py [hunk 2]
```diff
@@ -4,14 +4,17 @@
...
         super().__init__()
         self.show_divider = show_divider
+        self.is_inline = is_inline
 
     def map_gov_params(self, field, **kwargs):
         label_class = "govuk-fieldset__legend--l"
 
         params = super().map_gov_params(field, **kwargs)
+        if self.is_inline:
+            params["classes"] = "govuk-radios--inline"
```
Comment (BenMillar-MOJ) [inline app/categories/widgets.py:17]: This is brilliant, thank you for adding this!

File: app/categories/domestic_abuse/forms.py [hunk 3]
```diff
@@ -1,27 +1,5 @@
-from app.categories.forms import QuestionForm
-from wtforms import RadioField
-from app.categories.widgets import CategoryRadioInput
-from wtforms.validators import InputRequired
+from app.categories.forms import SafeguardingQuestionForm
 
 
-class AreYouAtRiskOfHarmForm(QuestionForm):
+class AreYouAtRiskOfHarmForm(SafeguardingQuestionForm):
```
Comment (BenMillar-MOJ) [inline app/categories/domestic_abuse/forms.py:4]: Can we rename this class, as the question no longer asks if the user is at risk of harm.

Can I suggest `WorriedAboutSomeonesSafetyForm`? Just so it's clear what this question is asking.

**Review by BenMillar-MOJ — CHANGES_REQUESTED** [review]
Looks really good! I have left a few small comments.

File: app/categories/forms.py (hunk 1 above)
Comment (said-moj) [reply app/categories/forms.py]: Removed

File: app/categories/domestic_abuse/forms.py (hunk 3 above)
Comment (said-moj) [reply app/categories/domestic_abuse/forms.py]: Renamed to `WorriedAboutSomeonesSafetyForm`

**Review by TawneeOwl — APPROVED** [review]
LGTM

**Review by BenMillar-MOJ — APPROVED** [review]

**Pull request merged**